src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,108 @@
-/* eslint-disable max-len */
-import React from 'react';
+import cn from 'classnames';
+import { actions } from '../../features/currentTodo';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
 
 export const TodoList: React.FC = () => {
+  const { todos, currentTodo, filter } = useAppSelector(state => state);
+  const { query, status } = filter;
+  const { setTodo, removeTodo } = actions;
+  const dispatch = useAppDispatch();
+
+  const filteredTodos = todos.filter(({ title, completed }) => {
+    const hasQuery = RegExp(query.trim(), 'i').test(title);
+    let hasStatus = true;
+
+    if (status === 'active') {
+      hasStatus = !completed;
+    }
+
+    if (status === 'completed') {
+      hasStatus = completed;
+    }
+
+    return hasQuery && hasStatus;
+  });
+
+  const toggleCurrentTodo = (todo: Todo) => () => {
+    if (currentTodo?.id === todo.id) {
+      dispatch(removeTodo());
+    } else {
+      dispatch(setTodo(todo));
+    }
```

Comment:
check whether it works this way, but it seems more ergonomic this way to me
```suggestion
  dispatch(
    currentTodo?.id === todo.id
      ? removeTodo()
      : setTodo(todo)
  )
```

Reply:
I think it works and yes this is better, thanks